modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -716,6 +728,92 @@ public String getProcesses() throws ProcessCenterException {
+                        String subprocessPath =
+                                subprocessElement.getElementsByTagName("path").item(0).getTextContent();
+                        String subprocessId =
+                                subprocessElement.getElementsByTagName("id").item(0).getTextContent();
+
+                        JSONObject subprocess = new JSONObject();
+                        subprocess.put("name", subprocessName);
+                        subprocess.put("path", subprocessPath);
+                        subprocess.put("id", subprocessId);
+                        subprocessArray.put(subprocess);
+                    }
+                }
+
+                if(successorElements.getLength() != 0){
+                    for(int i = 0 ; i < subprocessElements.getLength() ; i++){
+                        Element successorElement = (Element) successorElements.item(i);
+                        String successorName = successorElement.getElementsByTagName("name").item(0).getTextContent();
+                        String successorPath =
+                                successorElement.getElementsByTagName("path").item(0).getTextContent();
+                        String successorId =
+                                successorElement.getElementsByTagName("id").item(0).getTextContent();
+
+                        JSONObject successor = new JSONObject();
+                        successor.put("name", successorName);
+                        successor.put("path", successorPath);
+                        successor.put("id", successorId);
+                        successorArray.put(successor);
+                    }
+                }
+
+                if(predecessorElements.getLength() != 0){
+                    for(int i = 0 ; i < predecessorElements.getLength() ; i++){
+                        Element predecessorElement = (Element) predecessorElements.item(i);
+                        String predecessorName = predecessorElement.getElementsByTagName("name").item(0).getTextContent();
+                        String predecessorPath =
+                                predecessorElement.getElementsByTagName("path").item(0).getTextContent();
+                        String predecessorId =
+                                predecessorElement.getElementsByTagName("id").item(0).getTextContent();
+
+                        JSONObject predecessor = new JSONObject();
+                        predecessor.put("name", predecessorName);
+                        predecessor.put("path", predecessorPath);
+                        predecessor.put("id", predecessorId);
+                        predecessorArray.put(predecessor);
+                    }
+                }
+                resourceString = conObj.toString();
+            }
+        }catch (Exception e) {
+            log.error("Failed to fetch Successor Predecessor and Subprocess information: " + resourcePath);
```

Comment:
better if log throwable also
log.error("Failed to fetch Successor Predecessor and Subprocess information: " + resourcePath, e);